framework/src/media/audio/audio_manager.c

```diff
@@ -489,22 +498,24 @@ int start_audio_stream_in(void *data, uint32_t frames)
 	int ret;
 	int retry = AUDIO_STREAM_RETRY_COUNT;
 	struct pcm *pcm;
-
-	// Todo: Need to check g_actual_audio_out_card_id index.
-	audio_card_info_t *cur_card = &g_audio_in_cards[g_actual_audio_in_card_id];
+	audio_card_info_t *cur_card;
 
 	if ((ret = get_active_audio_device_pcm(&pcm, INPUT)) != AUDIO_MANAGER_SUCCESS) {
 		return ret;
 	}
 
-	if (g_audio_in_cards[g_actual_audio_in_card_id].status == AUDIO_CARD_PAUSE) {
+	pthread_mutex_lock(&(g_audio_in_cards[g_actual_audio_in_card_id].card_mutex));
```

Comment:
Is there any difference compare with direct initiate?

Reply:
Hmm.. What do you mean about "direct initiate"...?  pthread_mutex_lock should be called at line 502..?